### PR TITLE
Added try to options to avoid errors when enabled is false (#19)

### LINKED ISSUE #19: Enabled set to false results in errors
## Describe the Bug
When setting enabled to false you get errors in the `aws_vpc_peering_connection_options.accepter` and the `aws_vpc_peering_connection_options.requester`. The errors seem to be because these two resources are missing `count`.

## Expected Behavior
When I set enabled to false I expect the module to do nothing.

## Steps to Reproduce
1. Set enabled to false with other variables filled with empty strings
2. Run `terraform plan` (no errors should be shown)
3. Run `terraform apply` (see errors)

## Error Message
```
Error: error reading VPC Peering Connection (): InvalidVpcPeeringConnectionId.Malformed: Invalid id: "" (expecting "pcx-...")
        status code: 400, request id: c62fad7b-fc41-45d5-9fcb-6b97d7836638

  on .terraform/modules/vpc_peering_2/accepter.tf line 105, in resource "aws_vpc_peering_connection_options" "accepter":
 105: resource "aws_vpc_peering_connection_options" "accepter" {



Error: error reading VPC Peering Connection (): InvalidVpcPeeringConnectionId.Malformed: Invalid id: "" (expecting "pcx-...")
        status code: 400, request id: fc8948ec-92c9-4901-834d-9e6263055096

  on .terraform/modules/vpc_peering_2/requester.tf line 120, in resource "aws_vpc_peering_connection_options" "requester":
 120: resource "aws_vpc_peering_connection_options" "requester" {
```

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: Ubuntu 18.04.4
 - Code Version: 0.7.0 (but I see the missing count in master)

## Additional Context
I'm trying to loop through a list of VPC Peers but due to count in modules not being supported I've created three modules that I've set to `enabled=false` if the list of VPC Peers doesn't have a second or third element.

```
locals {
  # A dirty hack to support the below module hack
  vpc_peers = [for index in [0, 1, 2] : {
    peer_vpc_id    = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.vpc_id : ""
    peer_region    = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.region : ""
    peer_vpc_name  = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.vpc_name : ""
    requester_role = (length(var.vpc_peers) > index) ? var.vpc_peers[index].requester_role : ""
    accepter_role  = (length(var.vpc_peers) > index) ? var.vpc_peers[index].accepter_role : ""
    enabled        = length(var.vpc_peers) > index
  }]
}

# A dirty hack as count and for_each is not supported for modules until 0.13 (currently 0.12)
module "vpc_peering_1" {
  source    = "git::https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account.git?ref=0.7.0"
  namespace = terraform.workspace
  stage     = var.env
  name      = "${data.terraform_remote_state.this.outputs.vpc_name}-to-${local.vpc_peers[0].peer_vpc_name}"
  enabled   = local.vpc_peers[0].enabled

  requester_aws_assume_role_arn             = local.vpc_peers[0].requester_role
  requester_region                          = var.aws_region
  requester_vpc_id                          = data.terraform_remote_state.this.outputs.vpc_id
  requester_allow_remote_vpc_dns_resolution = true

  accepter_aws_assume_role_arn             = local.vpc_peers[0].accepter_role
  accepter_region                          = var.aws_region
  accepter_vpc_id                          = local.vpc_peers[0].peer_vpc_id
  accepter_allow_remote_vpc_dns_resolution = true
}

# A dirty hack as count and for_each is not supported for modules until 0.13 (currently 0.12)
module "vpc_peering_2" {
  source    = "git::https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account.git?ref=0.7.0"
  namespace = terraform.workspace
  stage     = var.env
  name      = "${data.terraform_remote_state.this.outputs.vpc_name}-to-${local.vpc_peers[0].peer_vpc_name}"
  enabled   = local.vpc_peers[1].enabled

  requester_aws_assume_role_arn             = local.vpc_peers[1].requester_role
  requester_region                          = var.aws_region
  requester_vpc_id                          = data.terraform_remote_state.this.outputs.vpc_id
  requester_allow_remote_vpc_dns_resolution = true

  accepter_aws_assume_role_arn             = local.vpc_peers[1].accepter_role
  accepter_region                          = var.aws_region
  accepter_vpc_id                          = local.vpc_peers[1].peer_vpc_id
  accepter_allow_remote_vpc_dns_resolution = true
}
```

### PR DESCRIPTION
**Pull request by JamesTimms**
## what
* Added count to accepter and requester options to prevent them from being created when enabled is false.

## why
* When enabled is false count is set to 0 and is used to skip most resources except these two options. When these options aren't skipped the module fails.

## references
* `closes #19 `

### CODE REVIEW

**Comment by JamesTimms** [comment]
I added a fix found for a related issue I explained in #19 which only effected Terraform 0.12.11+

**Comment by Gowiem** [comment]
@JamesTimms Can you fix your conflict?

**Comment by JamesTimms** [comment]
@Gowiem I've merged master but saw someone made a similar fix to one of the issues using try(). So I've changed my fix to use try() instead of length() and a ternary. It seems to work fine for me as long as I'm on a version on Terraform that supports try().

**Comment by JamesTimms** [comment]
@nitrocode & @Gowiem is this ok to merge now or do you need a few more changes?

**Comment by aknysh** [comment]
@JamesTimms  thanks for the PR.
Looks like in the final version you removed `count  = local.count` and the final changes are not reflected in the PR description.

I think the `count` needs to be added anyway to prevent the resources from being created when `enabled=false`.
You final changes are ok, they make TF to not throw errors when `data.aws_subnet_ids.accepter.*.ids` does not exist.

Please update.

**Comment by JamesTimms** [comment]
@aknysh,
I think someone added the count into a different PR that's made its way into master as the count is still there but not showing up in the diff.

Which lines exactly do you think I've removed count from? The commit [here](https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account/pull/20/commits/0c0b3f98f86b78c0eb119d419dfbe1b96dca9559), shows me adding them but then looking at the files ([here ](https://github.com/JamesTimms/terraform-aws-vpc-peering-multi-account/blob/50e1cc132ea60c2876fba7f9ad39eb1877252f7b/requester.tf#L121)and [here](https://github.com/JamesTimms/terraform-aws-vpc-peering-multi-account/blob/50e1cc132ea60c2876fba7f9ad39eb1877252f7b/accepter.tf#L106)) you can see they're still there despite not showing up in the diff.

**Comment by JamesTimms** [comment]
Is this likely to get merged?

**Comment by joe-niland** [comment]
@aknysh I think the count change is not showing in this PR because it was added here: https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account/commit/071e574b62abb4bf9c20820c114096cf25b5cc9e

@JamesTimms given the above, maybe good to update the PR description to match the remaining changes in this PR.

File: accepter.tf [hunk 1]
```diff
@@ -58,7 +58,7 @@ data "aws_subnet_ids" "accepter" {
 }
 
 locals {
-  accepter_subnet_ids       = distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids)))
+  accepter_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids))), "")
```
Comment (aknysh) [inline accepter.tf:61]:
```suggestion
  accepter_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids))), [])
```

File: requester.tf [hunk 2]
```diff
@@ -87,7 +87,7 @@ data "aws_subnet_ids" "requester" {
 }
 
 locals {
-  requester_subnet_ids       = distinct(sort(flatten(data.aws_subnet_ids.requester.*.ids)))
+  requester_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.requester.*.ids))), "")
```
Comment (aknysh) [inline requester.tf:90]:
```suggestion
  requester_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.requester.*.ids))), [])
```

File: requester.tf [hunk 3]
```diff
@@ -131,7 +131,7 @@ resource "aws_vpc_peering_connection_options" "requester" {
 }
 
 locals {
-  requester_aws_route_table_ids           = distinct(sort(data.aws_route_table.requester.*.route_table_id))
+  requester_aws_route_table_ids           = try(distinct(sort(data.aws_route_table.requester.*.route_table_id)), "")
```
Comment (aknysh) [inline requester.tf:134]:
```suggestion
  requester_aws_route_table_ids           = try(distinct(sort(data.aws_route_table.requester.*.route_table_id)), [])
```

although `""` will work as well, let's use the same data type

**Review by aknysh — COMMENTED** [review]
@JamesTimms thanks again, just a few nitpicks

**Comment by aknysh** [comment]
> @aknysh I think the count change is not showing in this PR because it was added here: [071e574](https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account/commit/071e574b62abb4bf9c20820c114096cf25b5cc9e)
> 
> @JamesTimms given the above, maybe good to update the PR description to match the remaining changes in this PR.

Agree with @joe-niland , please update the PR description

**Comment by JamesTimms** [comment]
> @JamesTimms thanks again, just a few nitpicks

I've updated to use empty arrays, including a related change that got into master via some other change. Let me know if this is good enough to merge!

**Review by joe-niland — APPROVED** [review]

**Comment by joe-niland** [comment]
Thanks for your contribution @JamesTimms !

**Pull request merged**